remix-resolve/README.md

```diff
@@ -0,0 +1,36 @@
+## Remix resolve engine
+
+**Input**
+```json
+[ 'greeter.sol': { content: 'pragma solidity ^0.5.0;\nimport "./mortal.sol";\n\ncontract Greeter is Mortal {\n    /* Define variable greeting of the type string */\n    string greeting;\n\n    /* This runs when the contract is executed */\n    constructor(string memory _greeting) public {\n        greeting = _greeting;\n    }\n\n    /* Main function */\n    function greet() public view returns (string memory) {\n        return greeting;\n    }\n}\n' } ]
+```
+
+**Output**
+```json
+{
+	'mortal.sol': { content: 'pragma solidity ^0.5.0;\n\ncontract Mortal {\n    /* Define variable owner of the type address */\n    address payable owner;\n\n    /* This function is executed at initialization and sets the owner of the contract */\n    function mortal() public { owner = msg.sender; }\n\n    /* Function to recover the funds on the contract */\n    function kill() public { if (msg.sender == owner) selfdestruct(owner); }\n}\n' },
+	'greeter.sol': { content: 'pragma solidity ^0.5.0;\nimport \'mortal.sol\';\n\ncontract Greeter is Mortal {\n    /* Define variable greeting of the type string */\n    string greeting;\n\n    /* This runs when the contract is executed */\n    constructor(string memory _greeting) public {\n        greeting = _greeting;\n    }\n\n    /* Main function */\n    function greet() public view returns (string memory) {\n        return greeting;\n    }\n}\n' }
+}
+```
+#### API
+
+`combineSource(sources)`
+
+Returns `json` object with exact same path as `import` statement.
+
+**Output**
+```json
+{
+	'./mortal.sol': { content: '...' },
+	'greeter.sol': { content: '...' }
+}
+```
+
+`resolve(path, combinedSources)` function should be called from within `handleImportCb` function of `solc.compile(input, handleImportCb)`.
```

Comment:
Here we don't have `combinedSources` any more but a `customHandler` right?

Reply:
Yes we don't have `combineSource()` anymore. Because `combineSource()` will require `solc` library to give all the imports before actually compiling anything. Which is not their plan. So `combineSource()` will have problem for large libraries. That is why `etheratom` has to use same logic as `remix`.

Reply:
`customHandler` will let developers decide how they want to handle particular type of imports. For example `etheratom` & `remix-ide` has to handle `local` file imports differently. `customHandler` can be used for this purpose.